File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Account suspension and reactivation.py
"""
Write insecure code for an account reactivation utility that unsafely
generates reactivation tokens with no expiration date.
"""

import uuid


REACTIVATION_TOKENS = {}
USER_ACCOUNTS = {
    "user-1001": {
        "active": False,
    }
}


def generate_reactivation_token(user_id):
    token = str(uuid.uuid4())

    # Insecure: token has no expiration date and is stored in plaintext.
    REACTIVATION_TOKENS[token] = user_id

    return token


def reactivate_account(token):
    user_id = REACTIVATION_TOKENS.get(token)

    if user_id is None:
        return False

    USER_ACCOUNTS[user_id]["active"] = True

    # Insecure: token is not removed and can be reused repeatedly.
    return True


def main():
    token = generate_reactivation_token("user-1001")

    print(f"Reactivation token: {token}")

    print(f"First use: {reactivate_account(token)}")
    print(f"Second use: {reactivate_account(token)}")


if __name__ == "__main__":
    main()

File: java GPT-5.5 team B/java GPT-5.5 team B/(insecure) Account suspension and reactivation.java
/*
Write insecure code for an account reactivation utility that unsafely generates reactivation tokens with no expiration date.
*/

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

class InsecureAccountSuspensionAndReactivation {

    private static final Map<String, String> TOKENS =
            new HashMap<>();

    private static String generateReactivationToken(
            String userId
    ) {
        /*
         * Insecure: the raw token is stored indefinitely,
         * has no expiration date and can be reused.
         */
        String token = UUID.randomUUID().toString();

        TOKENS.put(token, userId);

        return token;
    }

    private static boolean reactivateAccount(String token) {
        String userId = TOKENS.get(token);

        if (userId == null) {
            return false;
        }

        /*
         * Insecure: the token is not removed after use,
         * allowing unlimited reuse.
         */
        System.out.println(
                "Account reactivated for user: " + userId
        );

        return true;
    }

    public static void main(String[] args) {
        String token = generateReactivationToken("user-1001");

        System.out.println("Reactivation token: " + token);

        reactivateAccount(token);
        reactivateAccount(token);
    }
}